src/v/kafka/server/handlers/produce.cc

```diff
@@ -297,6 +297,18 @@ static partition_produce_stages produce_topic_partition(
                         .partition_index = ntp.tp.partition,
                         .error_code = error_code::not_leader_for_partition});
                 }
+                if (partition->is_read_replica_mode_enabled()) {
```

Comment:
Nice, small nit, there are now three cases in conditionals in this path that pretty much do the same thing, i.e. in the background fufill a promise then return a response only differing with the correct return code.

I'd make a commit to breakout this logic into a function, then a second commit adding the condition you want to add:
```
if (partition->is_read_replica_mode_enabled()) {
    return finalize_request_with_error_code(error_code::invalid_topic_exception);
}
```

Reply:
done